### PR TITLE
fix: support reading type arrow

### PR DESCRIPTION
**Pull request by machow**
Addresses #135

### CODE REVIEW

**Comment by machow** [comment]
Based on this comment https://github.com/rstudio/pins-python/issues/135#issuecomment-1161854818 switching "arrow" to be the go-to type, deprecating "feather".

**Comment by machow** [comment]
@juliasilge do you mind double checking this (either the code or sanity checking it does what we discussed!)?

I set pins to...

* pin_write and pin_read type "arrow"
* pin_read type "feather"
* pin_write "feather", but only after ...
  - printing a deprecation warning
  - switching the type to "arrow" (so the pin is technically saved as an arrow pin type)

File: pins/drivers.py [hunk 1]
```diff
@@ -123,11 +128,20 @@ def save_data(
...
+        #       we allow reading this type, but raise an error for writing.
         _assert_is_pandas_df(obj)
 
         obj.to_feather(final_name)
 
+    elif type == "feather":
+        _assert_is_pandas_df(obj)
+
+        raise NotImplementedError(
+            'Saving data as type "feather" no longer supported. Use type arrow instead.'
```
Comment (juliasilge) [inline pins/drivers.py:142]:
```suggestion
            'Saving data as type "feather" no longer supported. Use type "arrow" instead.'
```

**Review by juliasilge — APPROVED** [review]
This looks great! This behavior is a good scheme moving forward. 👍

**Pull request merged**